pkg/sources/kafka/reader.go

```diff
@@ -159,11 +157,6 @@ loop:
 	for i := int64(0); i < count; i++ {
 		select {
 		case m := <-ks.handler.messages:
-			kafkaSourceReadCount.With(map[string]string{
-				metrics.LabelVertex:        ks.vertexName,
-				metrics.LabelPipeline:      ks.pipelineName,
-				metrics.LabelPartitionName: strconv.Itoa(int(m.Partition)),
```

Comment:
Not having this means there won't be read metrics for each Kafka partition?

Reply:
Yes, I thought about it. Since this metric(`kafka_source_read_total`) is implemented at message level we have the partition number but cannot be implemented for `forwarder_data_read_total`. Should we keep this?

Reply:
Kafka partition is not the same partition number at message level.

Reply:
<img width="760" alt="Screenshot 2024-11-15 at 2 37 33 PM" src="https://github.com/user-attachments/assets/717becf8-6877-4dfc-a036-f493f067f437">
<img width="736" alt="Screenshot 2024-11-15 at 2 37 42 PM" src="https://github.com/user-attachments/assets/c4814bb8-676c-463c-955b-aae7d0572171">

`LabelPartitionName`: `m.Partition` could be 0, 1.. (kafka partition) in case of `kafka_source_read_total`
But in case of `forwarder_data_read_total`, we cannot distinguish read metrics based on kafka partition. Is my understanding correct?

Reply:
Can we use `Offset.PartitionIdx()` as partition?

Reply:
<img width="985" alt="Screenshot 2024-11-18 at 3 03 55 PM" src="https://github.com/user-attachments/assets/31a8dbb1-1b7e-4ced-ac19-dc8556e13ce8">

as discussed, brought read counter metrics for source inside for loop:
```
		metrics.ReadDataMessagesCount.With(map[string]string{
			metrics.LabelVertex:             df.vertexName,
			metrics.LabelPipeline:           df.pipelineName,
			metrics.LabelVertexType:         string(dfv1.VertexTypeSource),
			metrics.LabelVertexReplicaIndex: strconv.Itoa(int(df.vertexReplica)),
			metrics.LabelPartitionName:      strconv.Itoa(int(m.ID.Index)), // this is equal to readOffset.PartitionIdx() which is equal to m.Partition
		}).Inc()
```